jax/_src/numpy/polynomial.py

```diff
@@ -292,6 +295,22 @@ def polymul(a1, a2, *, trim_leading_zeros=False):
   val = convolve(a1, a2, mode='full')
   return val
 
+@_wraps(np.polydiv, lax_description=_LEADING_ZEROS_DOC)
+def polydiv(u, v, *, trim_leading_zeros=False):
+  _check_arraylike("polydiv", u, v)
+  u, v = _promote_dtypes_inexact(u, v)
+  m = len(u) - 1
+  n = len(v) - 1
+  scale = 1. / v[0]
+  q = zeros(max(m - n + 1, 1))
+  for k in range(0, m-n+1):
+    d = scale * u[k]
+    q = q.at[k].set(d)
+    u = u.at[k:k+n+1].add(-d*v)
+  if trim_leading_zeros:
+    return q, trim_zeros(where(abs(u)<1e-3, 0, u), trim='f')
```

Comment:
I think this is probably fine, but we may need to revisit this cutoff. I think numpy always converts to float64 and uses an `rtol=1E-14` cutoff. In JAX we may be computing in float32, bfloat16, etc. and perhaps we should change the cutoff depending on the width of the dtype.

Reply:
Since the PR needs a flake fix, can you also add a TODO here to revisit the tolerance depending on the dtype?

Reply:
No problem. Since I will use the ```jnp.polydiv``` myself, I will probably do some grid search to find the best tolerance here.

Reply:
I would probably choose some multiple of `jnp.finfo(dtype).eps` as a baseline.

Reply:
Just used the ```jnp.finfo(dtype).eps``` for cutoff and also added some code for ```dtype```.